cmd/tendermint/commands/light.go

```diff
@@ -25,10 +26,57 @@ import (
 )
 
 // LightCmd represents the base command when called without any subcommands
-var LightCmd = &cobra.Command{
-	Use:   "light [chainID]",
-	Short: "Run a light client proxy server, verifying Tendermint rpc",
-	Long: `Run a light client proxy server, verifying Tendermint rpc.
+func MakeLightCommand(conf *config.Config, logger log.Logger) *cobra.Command {
+	var (
+		listenAddr         string
+		primaryAddr        string
+		witnessAddrsJoined string
+		chainID            string
+		dir                string
+		maxOpenConnections int
+
+		sequential     bool
+		trustingPeriod time.Duration
+		trustedHeight  int64
+		trustedHash    []byte
+		trustLevelStr  string
+
+		logLevel  string
+		logFormat string
+
+		primaryKey   = []byte("primary")
+		witnessesKey = []byte("witnesses")
+	)
+
+	checkForExistingProviders := func(db dbm.DB) (string, []string, error) {
```

Comment:
(optional) I think we could probably leave `primaryKey` and `witnessKey`, and a couple of these helper functions, as top-level declarations. 

The keys are morally constant, and as best I can tell do not get changed. The functions don't close over any state, and having them at the top level might help make this big function a little easier to read.